### PR TITLE
Feature/flt strategies

### PR DESCRIPTION
**Pull request by bcanzanella**

### CODE REVIEW

File: shared/agent/src/providers/newrelic/anomalyDetection.ts [hunk 1]
```diff
@@ -9,76 +13,289 @@ interface NameValue {
...
+				const deploymentDate = new Date(deployment.seconds * 1000);
+				const now = new Date();
+				const timeDifference = now.getTime() - deploymentDate.getTime();
+				const daysDifference = Math.round(timeDifference / (1000 * 60 * 60 * 24));
+				this._dataTimeFrame = `SINCE ${daysDifference} days AGO`;
+				this._totalDays = daysDifference + parseInt(this._request.baselineDays as any);
+			}
+		}
+
+		const includeSpans = this._request.includeSpans || true;
```
Comment (bcanzanella) [inline shared/agent/src/providers/newrelic/anomalyDetection.ts:62]: won't this always be true?

File: shared/ui/Stream/CLMSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+					}
+					setUserPreference({
+						prefPath: ["clmSettings"],
+						value: {
+							["isChangeTrackingEnabled"]: isChangeTrackingEnabled,
+						},
+					});
+				})
+				.catch(ex => {
+					console.error("ERROR: GetDeploymentsRequestType", ex);
```
Comment (bcanzanella) [inline shared/ui/Stream/CLMSettings.tsx:77]: should this be user-visible? cc @eric-j

File: shared/ui/Stream/ObservabilityAnomaliesThroughputDropdown.tsx [hunk 3]
```diff
@@ -13,6 +12,7 @@ interface Props {
 	entityGuid?: string;
 }
 
+//@TODO Candidate for deletion
 export const ObservabilityAnomaliesThroughputDropdown = React.memo((props: Props) => {
```
Comment (bcanzanella) [inline shared/ui/Stream/ObservabilityAnomaliesThroughputDropdown.tsx:16]: if not used, can it be deleted?

File: shared/ui/Stream/ObservabilityAnomaliesErrorRateDropdown.tsx [hunk 4]
```diff
@@ -13,11 +12,13 @@ interface Props {
 	entityGuid?: string;
 }
 
+//@TODO: Candidate for deletion, probably do not need
```
Comment (bcanzanella) [inline shared/ui/Stream/ObservabilityAnomaliesErrorRateDropdown.tsx:15]: if not used, can it be deleted?

File: shared/ui/Stream/MethodLevelTelemetry/ObservabilityAnomalyPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,503 @@
...
+
+		loadData(derivedState.currentObservabilityAnomalyEntityGuid);
+	}, [derivedState.currentObservabilityAnomaly]);
+
+	if (
+		false
+		// derivedState.currentMethodLevelTelemetry.error &&
+		// derivedState.currentMethodLevelTelemetry.error.type === "NOT_ASSOCIATED" &&
+		// derivedState.currentMethodLevelTelemetry.repo
+	) {
```
Comment (bcanzanella) [inline shared/ui/Stream/MethodLevelTelemetry/ObservabilityAnomalyPanel.tsx:196]: TODO?

File: shared/agent/src/providers/newrelic/anomalyDetection.ts (hunk 1 above)
Comment (marcelofarias) [reply shared/agent/src/providers/newrelic/anomalyDetection.ts:62]: It's a moving target - this whole implementation needs some love before we move on to the next languages

File: shared/agent/src/providers/newrelic.ts [hunk 6]
```diff
@@ -1911,34 +1911,54 @@ export class NewRelicProvider extends ThirdPartyIssueProviderBase<CSNewRelicProv
...
+			if (request.includeDeployments && request.since) {
+				deployments = (
+					await this.getDeployments({
+						entityGuid: request.newRelicEntityGuid || entity!.entityGuid!,
+						since: request.since,
+					})
+				).deployments;
+			}
+
+			const entityGuid = entity?.entityGuid || request.newRelicEntityGuid;
```
[the remark this answers is not in the document]
Comment (marcelofarias) [reply shared/agent/src/providers/newrelic.ts:1953]: Favoring what's resolved (if resolved). There might be somebody calling it without  a newRelicEntityGuid. In any case, they should be the same.

File: shared/agent/src/providers/newrelic/clm/FLTCodeAttributeStrategy.ts [hunk 7]
```diff
@@ -0,0 +1,553 @@
...
+		const errorRatesConsolidated = [];
+		const sampleSizesMetric = sampleSizeResponse?.actor?.account?.metrics;
+		const sampleSizesSpan = sampleSizeResponse?.actor?.account?.spans;
+		const sampleSizesConsolidated = [];
+		for (const methodName of sampleSizeMap.keys()) {
+			const sampleSizeInfo = sampleSizeMap.get(methodName);
+			const sampleSizeMetricValue = sampleSizeInfo?.metric || 0;
+			const sampleSizeSpanValue = sampleSizeInfo?.span || 0;
+			const canUseMetric = true; // sampleSizeMetric > 30;
+			const canUseSpan = true; // sampleSizeSpan > 30;
```
[the remark this answers is not in the document]
Comment (marcelofarias) [reply shared/agent/src/providers/newrelic/clm/FLTCodeAttributeStrategy.ts:77]: Probably. Another moving target.

File: shared/ui/Stream/ObservabilityAnomaliesThroughputDropdown.tsx (hunk 3 above)
Comment (marcelofarias) [reply shared/ui/Stream/ObservabilityAnomaliesThroughputDropdown.tsx:16]: I think so - I believe @eric-j unified the other 2 in a single component. But again, this whole code will need some love once we settle on the actual UX and mechanics.

File: shared/ui/Stream/ObservabilityAnomaliesErrorRateDropdown.tsx (hunk 4 above)
Comment (eric-j) [reply shared/ui/Stream/ObservabilityAnomaliesErrorRateDropdown.tsx]: nuked

**Pull request merged**